base_layer/core/src/chain_storage/lmdb_db/lmdb_db.rs

```diff
@@ -629,22 +628,60 @@ impl LMDBDatabase {
         txn: &WriteTransaction<'_>,
         header: &BlockHeader,
         accum_data: &BlockHeaderAccumulatedData,
-    ) -> Result<bool, ChainStorageError>
+    ) -> Result<(), ChainStorageError>
     {
         if let Some(current_header_at_height) = lmdb_get::<_, BlockHeader>(txn, &self.headers_db, &header.height)? {
             let hash = current_header_at_height.hash();
-            if current_header_at_height.hash() != accum_data.hash {
+            if hash != accum_data.hash {
                 return Err(ChainStorageError::InvalidOperation(format!(
                     "There is a different header stored at height {} already. New header ({}), current header: ({})",
                     header.height,
                     hash.to_hex(),
                     accum_data.hash.to_hex()
                 )));
             }
-            return Ok(false);
+            return Err(ChainStorageError::InvalidOperation(format!(
+                "This header aready has been created. Header: {}",
```

Comment:
nit: Something like this may be better
```suggestion
                "The header at height {} already exists. Existing header hash: {}",
```